src/renderer/context/testSequencerWS.context.tsx

```diff
@@ -81,34 +98,37 @@ export function TestSequencerWSProvider({
   // Run when a new WebSocket message is received (lastJsonMessage)
   useEffect(() => {
     if (!lastJsonMessage) return;
-
     const msg = BackendMsg.safeParse(lastJsonMessage);
     if (!msg.success) {
       console.error(msg.error);
       return;
+    } else {
+      console.log("Received message from backend: ", msg.data);
     }
 
     switch (msg.data.state) {
```

Comment:
questionable, each of these handlers could just be a store action (alternatively just turn the entire switch case into a store action that takes in a socket message)

Reply:
The switch statement will have to be redone at the same time as the state management, changing this makes things more complicated and confusing, because as it uses other hooks, half the logic is somewhere, and the other half is somewhere else.
I should have changed everything at the beginning instead of building on what already existed, but I didn't have the time due to the big scope / short deadlines